src/utils.js

```diff
@@ -15,18 +21,12 @@ import * as Errors from 'bitgo/dist/src/errors';
  */
 export async function recoverWithKeyPath(baseCoin, recoveryParams) {
   const userKeyPaths = ['/0/0', '/0'];
-  let recoveryPrebuild;
 
   for (const path of userKeyPaths) {
-    recoveryParams['userKeyPath'] = path;
-
     try {
-      recoveryPrebuild = await baseCoin.recover(recoveryParams);
       // if we already have a recovery result, then it means the current path we try
       // is the valid user path, and we can return and exit the iteration loop
-      if (recoveryPrebuild) {
-        return recoveryPrebuild;
-      }
+      return await baseCoin.recover(withCustomKeyPath(recoveryParams, path));
```

Comment:
Not sure but you might not need `await` here. This function is already marked `async` and if `baseCoin.recover()` returns a promise then I think you're ok.

Reply:
`await` is necessary here in order for the `try` block in this function to catch an exception thrown during promise resolution

Reply:
👍